src/worker/move-contact-worker.ts

```diff
@@ -0,0 +1,156 @@
+import axios from 'axios';
+import { spawn } from 'child_process'; 
+import { Worker, Job, DelayedError } from 'bullmq';
+
+import Auth from '../lib/authentication';
+import { redisConnection } from '../shared/queue-config';
+import { DateTime } from 'luxon';
+
+export interface MoveContactData {
+  parentId: string;
+  contactId: string;
+  sessionToken: string;
+  instanceUrl: string;
+}
+
+export type JobResult = { success: boolean; message: string };
+
+export class MoveContactWorker {    
+  private readonly DELAY_IN_MILLIS = 60 * 60 * 1000;       // 60 minutes
+  private readonly MAX_TIMEOUT_IN_MILLIS = 60 * 60 * 1000; // 60 minutes
+  private readonly MAX_CONCURRENCY = 1;                    // Limit concurrency to 1 job at a time
+  private readonly MAX_SENTINEL_BACKLOG = 7000;            // ensure we don't take down the server
+  
+  constructor(private queueName: string) {
+    this.initializeWorker();
+  }
+
+  private initializeWorker() {
+    return new Worker(
+      this.queueName, 
+      this.handleJob, 
+      { connection: redisConnection, concurrency: this.MAX_CONCURRENCY }
+    );
+  }
+
+  private handleJob = async (job: Job, processingToken?: string): Promise<boolean> => {
+    const jobData: MoveContactData = job.data;
+
+    // Ensure server availability
+    if (await this.shouldPostpone(jobData)) {
+      await this.postpone(job, processingToken);
+      throw new DelayedError();
+    }
+
+    const result = await this.moveContact(jobData);
+    if (!result.success) {
+      job.log(`[${new Date().toISOString()}]: ${result.message}`);
+      const errorMessage = `Job ${job.id} failed with the following error: ${result.message}`;
+      console.error(errorMessage);
+      throw new Error(errorMessage);
```

Comment:
at some point it might be nice to handle errors  a bit better. maybe not in this PR, but what if cht-conf has an error - the most common scenario here is losing connectivity with the server during a heavy move. should the job be retried?


i'll make a quick comment about aborted/canceled jobs or generally handling any case where a contact's move is interrupted

* the good news is that if a move is interrupted, regardless of the state cht-conf can be rerun. it will aggregate all documents (those in the new position already and those in the old position) and move them all again. this is really an incredible win - if a job fails, it just needs to be retried.
* the bad news is that cht-user-management won't let you retry an aborted job if the top-level contact successfully moves. it will give an error like "A already has B as parent" or whatever. 

It's going to happen, and we're going to need to deal with it at some point.

Reply:
This is really important, the way I was dealing with it is to see from the `Board`, the issue with the failed job (from the logs).
If it has failed due to the lost of connectivity, then I just `retry` from the `Board`.

But we can handle this properly in another PR.